calico/election.py

```diff
@@ -173,49 +175,77 @@ def _become_master(self):
                 master.
                 Any other error from etcd is not caught in this routine.
         """
-        id_string = "%s:%d:%s" % (
-            self._server_id, os.getpid(), int(time.time())
-        )
 
         try:
             self._etcd_client.write(self._key,
-                                    id_string,
+                                    self.id_string,
                                     ttl=self._ttl,
-                                    prevExists=False,
+                                    prevExist=False,
                                     timeout=self._interval)
-
         except Exception as e:
             # We could be smarter about what exceptions we allow, but any kind
             # of error means we should give up, and safer to have a broad
             # except here.  Since we expect to hit this in the mainline, we
             # don't log out the stack trace.
-            _log.info("Failed to become elected master - key %s: %r",
-                      self._key, e)
+            self._log_exception("become elected master", e)
```

Comment:
Actually, I think _log_exception sometimes does log out the stack trace, doesn't it? Then the comment isn't quite correct.